### ISSUE TITLE
Applying Papagayo lip sync data sometimes fails

### ISSUE THREAD

**Issue by artisteacher** [opentoonz/opentoonz#2072]
### --Issue Summary--

In certain cases, using the "Apply Lip Sync Data to Column" to apply lip sync data from Papagayo fails to have any effect.
### --Video or Image Reference--

Problem 1:
![screen shot 2018-06-13 at 8 11 26 am](https://user-images.githubusercontent.com/21159116/41350562-93ebf6a4-6ee1-11e8-851b-97a6e69317b7.png)

Problem 2:
![thiswillfail](https://user-images.githubusercontent.com/21159116/41351102-58664e70-6ee3-11e8-8840-c8cec8592b3b.png)
[incorrectphonemeplacement.dat.zip](https://github.com/opentoonz/opentoonz/files/2098065/incorrectphonemeplacement.dat.zip)
![screen shot 2018-06-13 at 8 27 29 am](https://user-images.githubusercontent.com/21159116/41351199-a864c5be-6ee3-11e8-89aa-53cd7c6c1f15.png)

### --Steps to reproduce--

There are 2 related issues:

1. If an empty cell in the column is selected before the column header, there is a warning message "unable to apply lip sync data to this column type" - even though the level in the column is the correct type. After closing the warning dialog, the lip sync window opens but is not usable.

2. Certain arrangements of the frame numbers in the .dat file from Papagayo makes the lip-sync fail:
I've had a few students run into this and the problem likely starts from user error on the Papagayo end. Having the first phoneme in Papagayo begin after the beginning of the first word causes the lip-sync to fail. Looking at some of the .dat files in a text editor, I was able to adjust the problem by fixing the numbering - it looks like having the frame numbers out of order (i.e. 1, 0, 3 or 4, 2, 6) is a problem. Also, the first frame in the Papagayo .dat file is usually -1. If the next frame is 1, then the lip-sync fails. Changing the second number to 2 allows the lip-sync data to apply as expected.
### --Actual Results--

The application of lip-sync data failed and no change occurred in the column.
### --Expected Results--

1. Since the lip-sync data applies to the column, I'd expect there to be no warning when there is an applicable level type in the column. When there is a warning, the lip-sync window should not open upon closing the warning.

2. It's expected that the data will apply to the column & that the mouth drawings will be placed accordingly in the column. If there's an uncorrectable problem with the .dat file, then there should at least be an error message.
### --System Information--
+ OpenToonz Version:

RC & nightly builds from late March
+ Operating System:

macOS 10.11.6

+ Other Device Info (Optional):

* RAM Size  4GB, 8GB, 16GB:
4GB
* Graphics Tablet:

---

**Comment by artisteacher** [comment]
At least in OS X, I've been able to add lip sync data to columns inside a sub-xsheet without issue. Have you tried recently?

**Comment by DarrenTAnims** [comment]
No, not recently. I think my version of OTX is from around a month or 2 ago.

**Comment by artisteacher** [comment]
When you are able to, you might want to try with the RC.

**Comment by merlin2v** [comment]
for windows, though i don't know if this is the same problem or not, rather than a neat error message,it just freezes.
![image](https://user-images.githubusercontent.com/25653109/45260990-17826080-b3ab-11e8-8c1c-d63a022cbe89.png)
if you try uploading a .dat file it will sit there and do nothing until you fill up the action queue by trying to click stuff(Which makes windows realize that the program is not responding) 
_Note: I did find the Papagayo-NG to be a bit buggy._

**Comment by artisteacher** [comment]
That looks like something else. So it freezes after adding the .dat file? What version of Windows are you using?

**Comment by artisteacher** [comment]
As a workaround, loading a .dat file generated from [Rhubarb](https://github.com/DanielSWolf/rhubarb-lip-sync) works without issue. So for mac users that are comfortable working in Terminal, Rhubarb might be a better option than Papagayo.

**Comment by johnosbb** [comment]
I just spent an entire day tearing my hair out before finding this post. In my case, it failed silently until I corrected a sequencing error in the .dat file

**Comment by artisteacher** [comment]
That's frustrating. What version of Papagayo did you use?

**Comment by johnosbb** [comment]
> That's frustrating. What version of Papagayo did you use?

It was 2.0b

**Comment by RodneyBaker** [comment]
This won't resolve any issue with Papagayo but currently nightly releases have Rhubarb lipsync support.
REF:  Rhubarb integration (Auto Lip-sync) #4261

**Comment by artisteacher** [comment]
I've given up on Papagayo at this point as Papagayo NG doesn't have a macOS release. Rhubarb does a great job getting the initial lip-sync. It's easy enough to make minor refinements if it ends up being needed.

**Comment by RodneyBaker** [comment]
Not sure how this applies but (for Rhubarb)... REF:  #6319 

As the author of this report has moved to Rhubarb will convert this to a Discussion so those interested in discussing Papagayo can delve further.
